### PR TITLE
Ixspd1 1805  Implemented payout event airdrop page

### PR DESCRIPTION
**Pull request by kapildave07**
## Description

 Implemented payout event airdrop page and 

## Changes

- Implemented payout event airdrop page
- Updated payout event creation to allow two modes  (Claim, airdrop)
-

## Attached Links

1. Jira ticket: 
https://investax.atlassian.net/browse/IXSPD1-1805
https://investax.atlassian.net/browse/IXSPD1-1806

2. Documents:

## Checklist

- [x] Manual tests
- [x] The pull request doesn't affect existing feature

## For reviewers

- Ensure new code doesn't break existing features
- Reviewer shouldn't merge if there are pending unresolved comments

### CODE REVIEW

File: src/components/Uploader/index.tsx [hunk 1]
```diff
@@ -29,102 +33,188 @@ export interface UploaderProps {
...
+      reader.onload = (event) => {
+        const text = event.target?.result as string
+        uploadCSVFile(text)
+      }
+      reader.readAsText(file)
+    }
+  }
+
+  const uploadCSVFile = (csvText: string) => {
+    const parsed = Papa.parse<string[]>(csvText, { delimiter: ',', skipEmptyLines: true }).data
```
Comment (thi-investax) [inline src/components/Uploader/index.tsx:73]: @kapildave07 does the csv work if it has a header 
`address,amount`?

for example:
address,amount
0xx.....,0.5

Comment (kapildave07) [reply src/components/Uploader/index.tsx:73]: @thi-investax that will not work here is the csv example

image

Comment (thi-investax) [reply src/components/Uploader/index.tsx:73]: what if the the csv file has a header? 
address,amount, it should work too

Comment (kapildave07) [reply src/components/Uploader/index.tsx:73]: no that will not work should we handle that too? Currently we are validating values only not headers.

**Comment by thi-investax** [comment]
image
could we move event name above memo? @kapildave07

File: src/components/Uploader/index.tsx [hunk 2]
```diff
@@ -29,102 +33,211 @@ export interface UploaderProps {
...
+    let hasError = false
+
+    if (parsed.length === 0) {
+      showError('Invalid file: The file is empty.')
+      return
+    }
+
+    const [headers, ...rows] = parsed
+
+    if (headers[0].trim() !== 'Wallet Address' || headers[1].trim() !== 'Amount') {
```
Comment (thi-investax) [inline src/components/Uploader/index.tsx:87]: should change to walletAddress,amount. Header shouldn't include space

**Comment by thi-investax** [comment]
I got this error when click confirm 
image

File: src/components/Uploader/index.tsx (hunk 2 above)
Comment (kapildave07) [reply src/components/Uploader/index.tsx]: @thi-investax resolved

**Pull request merged**